src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/CSharp/DefaultTagHelperDescriptorFactory.cs

```diff
@@ -242,107 +243,101 @@ private static void ConfigureDictionaryBoundAttribute(
         BoundAttributeDescriptorBuilder builder,
         IPropertySymbol property,
         INamedTypeSymbol containingType,
-        AttributeData attributeNameAttribute,
+        AttributeData? attributeNameAttribute,
         string attributeName,
         bool hasPublicSetter)
     {
-        string dictionaryAttributePrefix = null;
-        var dictionaryAttributePrefixSet = false;
+        string? dictionaryAttributePrefix = null;
 
         if (attributeNameAttribute != null)
         {
-            foreach (var argument in attributeNameAttribute.NamedArguments)
+            foreach (var (name, argument) in attributeNameAttribute.NamedArguments)
             {
-                if (argument.Key == TagHelperTypes.HtmlAttributeName.DictionaryAttributePrefix)
+                if (name == TagHelperTypes.HtmlAttributeName.DictionaryAttributePrefix)
                 {
-                    dictionaryAttributePrefix = (string)argument.Value.Value;
-                    dictionaryAttributePrefixSet = true;
```

Comment:
Is this a behavior change? Previously we would consider the prefix set even for `null` value.

Reply:
Yea, I think it may be. I'll put that back.